File: src/main/java/org/goafabric/personservice/persistence/entity/AddressEo.java
package org.goafabric.personservice.persistence.entity;

import jakarta.persistence.*;
import org.goafabric.personservice.persistence.extensions.AuditTrailListener;


@Entity
@Table(name="address")
@EntityListeners(AuditTrailListener.class)
public class AddressEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String street;
    private String city;

    @Version //optimistic locking
    private Long version;

    public AddressEo(String id, String street, String city, Long version) {
        this.id = id;
        this.street = street;
        this.city = city;
        this.version = version;
    }

    AddressEo() {
    }

    public String getId() {
        return id;
    }

    public String getStreet() {
        return street;
    }

    public String getCity() {
        return city;
    }

    public Long getVersion() {
        return version;
    }
}

File: src/main/kotlin/org/goafabric/personservice/persistence/entity/AddressEo.kt
package org.goafabric.personservice.persistence.entity

import jakarta.persistence.*
import org.goafabric.personservice.persistence.extensions.AuditTrailListener
import org.goafabric.personservice.persistence.extensions.KafkaPublisher

@Entity
@Table(name = "address")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class AddressEo (
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String?,

    var street: String,
    var city: String,

    @Version //optimistic locking
    var version: Long?
)

